NetCord/User.cs

```diff
@@ -2,43 +2,230 @@
+    ///         <description>
+    ///         Usernames cannot contain the following substrings: <c>@</c>, <c>#</c>, <c>:</c>, <c>```</c>, <c>discord</c>.
+    ///         </description>
+    ///     </item>
+    ///     <item>
+    ///         <description>
+    ///         Usernames cannot be: <c>everyone</c> or <c>here</c>.
+    ///         </description>
+    ///     </item>
+    /// </list>
+    /// Requires the <c>identify</c> OAuth2 scope.
+    /// </remarks>
     public string Username => _jsonModel.Username;
+
+    /// <summary>
+    /// The user's Discord-tag.
+    /// </summary>
+    /// <remarks>
+    /// Requires the <c>identify</c> OAuth2 scope.
+    /// </remarks>
     public ushort Discriminator => _jsonModel.Discriminator;
+
+    /// <summary>
+    /// The user's display name, if it is set. For bots, this is the application name. Restrictions:
+    /// </summary>
+    /// <remarks>
+    /// <list type="bullet">
+    ///     <item>
+    ///         <description>
+    ///         Must be between 1 and 32 characters long.
+    ///         </description>
+    ///     </item>
+    ///     <item>
+    ///         <description>
+    ///         Some zero-width and non-rendering characters are limited.
+    ///         </description>
+    ///     </item>
+    ///     <item>
+    ///         <description>
+    ///         Names are sanitized and trimmed of leading, trailing, and excessive internal whitespace.
+    ///         </description>
+    ///     </item>
+    /// </list>
+    /// Requires the <c>identify</c> OAuth2 scope.
+    /// </remarks>
     public string? GlobalName => _jsonModel.GlobalName;
+
+    /// <summary>
+    /// The user's avatar hash.
+    /// </summary>
+    /// <remarks>
+    /// Requires the <c>identify</c> OAuth2 scope.
+    /// </remarks>
     public string? AvatarHash => _jsonModel.AvatarHash;
+
+    /// <summary>
+    /// Whether the user belongs to an OAuth2 application.
+    /// </summary>
+    /// <remarks>
+    /// Requires the <c>identify</c> OAuth2 scope.
+    /// </remarks>
     public bool IsBot => _jsonModel.IsBot;
+
+    /// <summary>
+    /// Whether the user is an Official Discord System user (part of the urgent message system).
+    /// </summary>
+    /// <remarks>
+    /// Requires the <c>identify</c> OAuth2 scope.
+    /// </remarks>
     public bool? IsSystemUser => _jsonModel.IsSystemUser;
+
+    /// <summary>
+    /// Whether the user has two factor enabled on their account.
+    /// </summary>
+    /// <remarks>
+    /// Requires the <c>identify</c> OAuth2 scope.
+    /// </remarks>
     public bool? MfaEnabled => _jsonModel.MfaEnabled;
+
+    /// <summary>
+    /// The user's banner hash.
+    /// </summary>
+    /// <remarks>
+    /// Requires the <c>identify</c> OAuth2 scope.
+    /// </remarks>
     public string? BannerHash => _jsonModel.BannerHash;
+
+    /// <summary>
+    /// The user's banner color encoded as an integer representation of hexadecimal color code.
```

Comment:
NetCord handles it conversions automatically. I don't think the representation is worth noting.

Reply:
It's still internally stored in the hex color format (#000000), it's RGB, but still hex.

Reply:
It's stored in binary 🙃

Reply:
> a color may be specified as an [RGB](https://en.wikipedia.org/wiki/RGB_color_model) triplet, in [hexadecimal](https://en.wikipedia.org/wiki/Hexadecimal) format (a _hex triplet_) or according to its common English name in some cases.

From: https://en.wikipedia.org/wiki/Web_colors

Reply:
But it says about the # format. NetCord doesn't use this format to store the color value, it uses 3 bytes.

Reply:
Removed the indicator entirely for simplicity.